flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -701,12 +701,14 @@ public boolean contains(ConfigOption<?> configOption) {
 			if (this.confData.containsKey(configOption.key())) {
 				return true;
 			}
-			else if (configOption.hasDeprecatedKeys()) {
-				// try the deprecated keys
-				for (String deprecatedKey : configOption.deprecatedKeys()) {
-					if (this.confData.containsKey(deprecatedKey)) {
-						LOG.warn("Config uses deprecated configuration key '{}' instead of proper key '{}'",
-							deprecatedKey, configOption.key());
+			else if (configOption.hasFallbackKeys()) {
+				// try the fallback keys
+				for (FallbackKey fallbackKey : configOption.fallbackKeys()) {
+					if (this.confData.containsKey(fallbackKey.getKey())) {
+						if (fallbackKey.isDeprecated()) {
+							LOG.warn("Config uses deprecated configuration key '{}' instead of proper key '{}'",
+								fallbackKey.getKey(), configOption.key());
```

Comment:
The warning message could be factored out in one method.

Reply:
does it mean to extract the LOG statement to one method like below?

```java
private void warningDeprecated(...) {
   LOG.warn(...);
}
```

I wonder what value it provide or just style convenience to follow. I don't find another code block with a method just logging one statement.

Reply:
Oh I see the repeat, would do as suggested, sorry :P